src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -1250,10 +1253,14 @@ bool is_object_superset(
         // newer)
         return false;
     }
-    if (!is_object_pattern_properties_superset(ctx, older, newer)) {
-        // pattern properties checks are not compatible
-        return false;
-    }
+
+    // note: to match the behavior of the legacy software,
```

Comment:
nit: I would delete the code for `is_object_pattern_properties_superset` since it will probably go stale/unmaintained soon. If we want to reinstate it, we should be able to use git blame on this comment to find this implementation to kickstart redevelopment.

Reply:
done